src/node_url.cc

```diff
@@ -162,23 +170,17 @@ namespace url {
               } else {
                 value = value * 10 + number;
               }
-              if (value > 255)
-                goto end;
               pointer++;
               ch = pointer < end ? pointer[0] : kEOL;
+              if (value > 255)
+                goto end;
             }
-            if (dots < 3 && ch != '.')
-              goto end;
             *piece_pointer = *piece_pointer * 0x100 + value;
-            if (dots & 0x1)
+            numbers_seen++;
+            if (numbers_seen == 2 || numbers_seen == 4)
               piece_pointer++;
-            if (ch != kEOL) {
-              pointer++;
-              ch = pointer < end ? pointer[0] : kEOL;
-            }
-            if (dots == 3 && ch != kEOL)
+            if (ch == kEOL && numbers_seen != 4)
```

Comment:
I think I see what you mean but in that case it can be moved to right after the loop, right?  And the `ch == kEOL` clause can be dropped because that's implied by `while (ch != kEOL)`.